answerking_app/utils/model_types.py

```diff
@@ -1,73 +1,49 @@
-from typing import TypedDict, Any
+from typing import Any
 
+from typing_extensions import (  # for Python <3.11 with (Not)Required
+    NotRequired,
+    TypedDict,
+)
 
-class IDType(TypedDict):
-    id: int
 
-
-class NewItemType(TypedDict):
+class ItemType(TypedDict):
+    id: NotRequired[int]
     name: str
     price: str
     description: str
+    retired: NotRequired[bool]
     stock: int
     calories: int
 
 
-class ItemType(IDType, NewItemType):
-    retired: bool
-
-
-class NewCategoryName(TypedDict):
-    name: str
-
-
-class NewCategoryItems(TypedDict):
-    items: list[ItemType]
-
-
-class NewCategoryType(NewCategoryName, NewCategoryItems):
-    pass
-
-
-class CategoryType(IDType, NewCategoryType):
-    retired: bool
+class CategoryType(TypedDict):
+    id: NotRequired[int]
+    name: NotRequired[str]
```

Comment:
why is name a not required field?

Reply:
Because sometimes it's used partially without "name" in the tests, that's just for typing not the actual model